biscuit-auth/src/token/mod.rs

```diff
@@ -500,7 +500,7 @@ impl Biscuit {
         1 + self.blocks.len()
     }
 
-    pub(crate) fn block(&self, index: usize) -> Result<Block, error::Token> {
+    pub fn block(&self, index: usize) -> Result<Block, error::Token> {
```

Comment:
now that the printing methods are updated, this change can be removed as well, because it would expose `token::block::Block`, which is not part of the public API (still marked as public in the code, but it was removed from the top level exports a while ago).
The `Biscuit::context` method should be more useful here: https://docs.rs/biscuit-auth/latest/biscuit_auth/struct.Biscuit.html#method.context

Reply:
after that last change I think this will be ok to merge :)

Reply:
and also `biscuit_block_count()`

Reply:
probably not, it is better to align with the current rust API, where we encourage using the parsers and builders. The C API was designed along with the older rust API

Reply:
Is this refactored example is valid?
```rust
#[no_mangle]
pub unsafe extern "C" fn biscuit_block_context(
    biscuit: Option<&Biscuit>,
    block_index: u32,
) -> *mut c_char {
    if biscuit.is_none() {
        update_last_error(Error::InvalidArgument);
        return std::ptr::null_mut();
    }

    let biscuit = biscuit.unwrap();

    let context = biscuit.0.context();

    match context.get(block_index as usize) {
        None => std::ptr::null_mut(),
        Some(context) => match CString::new(context.clone().unwrap()) {
            Ok(s) => s.into_raw(),
            Err(_) => {
                update_last_error(Error::InvalidArgument);
                std::ptr::null_mut()
            }
        },
    }
}
```

Reply:
you'd want the following logic:
- if `get()` fails, you have the InvalidBlockId error
- if you get `Some(context)` but `context` is `None` (the block exists but had no context string, then return a null pointer without calling `update_last_error`

Reply:
Ok I've removed `count()` method for now and update the `biscuit_block_context` methode here:

[`08f7900` (#234)](https://github.com/biscuit-auth/biscuit-rust/pull/234/commits/08f79006e062c840d4e7ca4baa15034081749bdf)